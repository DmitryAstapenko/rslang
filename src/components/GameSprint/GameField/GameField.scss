.game-status {
  width: 100%;
  border-bottom: 1px solid black;
}

.game-button-container {
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 4px !important;
}

.game-field{
  width: 55%;
  max-width: 400px;
  text-align: center;
}
