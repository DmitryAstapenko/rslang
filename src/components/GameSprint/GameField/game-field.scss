.game-status {
  width: 100%;
  border-bottom: 1px solid black;
}
.game-button-container {
  display: flex;
  align-items: center;
  justify-content: center;
}
