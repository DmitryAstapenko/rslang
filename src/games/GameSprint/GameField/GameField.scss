.game-status {
  width: 100%;  
}

.game-button-container {
  margin-top: 1rem;
  display: flex;
  align-items: center;
  justify-content: center;  

  & button {
    margin: 0 1rem;
    width: 8rem;
    font-weight: bold;
  }
}

.game-field{
  padding: 1rem;
  width: 30rem;  
  text-align: center;
}
