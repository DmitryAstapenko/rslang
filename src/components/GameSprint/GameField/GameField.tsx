import React, { FunctionComponent } from "react";

import Card from "@material-ui/core/Card";
import {
  LooksOne,
  LooksTwo,
  Looks3,
  Looks4,
  CheckBox,
  CheckBoxOutlineBlank,
} from "@material-ui/icons";
import CardActions from "@material-ui/core/CardActions";
import CardContent from "@material-ui/core/CardContent";
import Button from "@material-ui/core/Button";
import Typography from "@material-ui/core/Typography";

import "./GameField.scss";

interface IGameFieldProps {
  currentWord: string;
  levelBonus: number;
  currentTranslate: string;
  onAnswerClick: (e: React.MouseEvent<HTMLButtonElement>) => void;
  currentChain: number;
}

// eslint-disable-next-line
const GameField: FunctionComponent<IGameFieldProps> = (props: IGameFieldProps) => {
  const {
    currentWord,
    levelBonus,
    currentTranslate,
    onAnswerClick,
    currentChain,
  } = props;
  let level;
  let chain;
  switch (levelBonus) {
    case 1:
      level = <LooksOne />;
      break;
    case 2:
      level = <LooksTwo />;
      break;
    case 3:
      level = <Looks3 />;
      break;
    case 4:
      level = <Looks4 />;
      break;
    default:
      level = <LooksOne />;
      break;
  }

  const emptyCheckBox = (
    <span>
      <CheckBoxOutlineBlank />
    </span>
  );

  const checkBox = (
    <span>
      <CheckBox />
    </span>
  );

  switch (currentChain) {
    case 0:
      chain = (
        <div>
          {emptyCheckBox}
          {emptyCheckBox}
          {emptyCheckBox}
        </div>
      );
      break;
    case 1:
      chain = (
        <div>
          {checkBox}
          {emptyCheckBox}
          {emptyCheckBox}
        </div>
      );
      break;
    case 2:
      chain = (
        <div>
          {checkBox}
          {checkBox}
          {emptyCheckBox}
        </div>
      );
      break;
    case 3:
      chain = (
        <div>
          {checkBox}
          {checkBox}
          {checkBox}
        </div>
      );
      break;

    default:
      break;
  }

  return (
    <Card className="game-field">
      <CardContent>
        <div className="game-status">
          {levelBonus === 4 ? checkBox : chain}
          <p>{`+ ${2 ** (levelBonus - 1) * 10} очков за слово`}</p>
        </div>
        <Typography gutterBottom variant="h5" component="h2">
          {level}
        </Typography>
        <Typography gutterBottom variant="h5" component="h2">
          {currentWord}
        </Typography>
        <Typography gutterBottom variant="h5" component="h2">
          {currentTranslate}
        </Typography>
      </CardContent>
      <CardActions className="game-button-container">
        <Button
          variant="contained"
          color="secondary"
          onClick={(e) => onAnswerClick(e)}
        >
          Неверно
        </Button>
        <Button
          variant="contained"
          color="primary"
          onClick={(e) => onAnswerClick(e)}
        >
          Верно
        </Button>
      </CardActions>
    </Card>
  );
};

export default GameField;
