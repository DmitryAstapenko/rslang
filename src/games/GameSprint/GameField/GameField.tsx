import React, { FunctionComponent } from "react";
import {
  LooksOneTwoTone as LooksOne,
  LooksTwoTwoTone as LooksTwo,
  Looks3TwoTone as Looks3,
  Looks4TwoTone as Looks4,
  CheckBoxTwoTone as CheckBox,
  CheckBoxOutlineBlankTwoTone as CheckBoxOutlineBlank,
} from "@material-ui/icons";
import Button from "@material-ui/core/Button";
import Typography from "@material-ui/core/Typography";
import "./GameField.scss";
import { Box, Divider, Paper } from "@material-ui/core";

interface IGameFieldProps {
  currentWord: string;
  levelBonus: number;
  currentTranslate: string;
  onAnswerClick: (e: React.MouseEvent<HTMLButtonElement>) => void;
  currentChain: number;
}

// eslint-disable-next-line
const GameField: FunctionComponent<IGameFieldProps> = (
  props: IGameFieldProps
) => {
  const {
    currentWord,
    levelBonus,
    currentTranslate,
    onAnswerClick,
    currentChain,
  } = props;
  let level;
  let chain;
  switch (levelBonus) {
    case 1:
      level = <LooksOne fontSize="large" />;
      break;
    case 2:
      level = <LooksTwo fontSize="large" />;
      break;
    case 3:
      level = <Looks3 fontSize="large" />;
      break;
    case 4:
      level = <Looks4 fontSize="large" />;
      break;
    default:
      level = <LooksOne fontSize="large" />;
      break;
  }

  const emptyCheckBox = (
    <span>
      <CheckBoxOutlineBlank fontSize="large" />
    </span>
  );

  const checkBox = (
    <span>
      <CheckBox fontSize="large" />
    </span>
  );

  switch (currentChain) {
    case 0:
      chain = (
        <div>
          {emptyCheckBox}
          {emptyCheckBox}
          {emptyCheckBox}
        </div>
      );
      break;
    case 1:
      chain = (
        <div>
          {checkBox}
          {emptyCheckBox}
          {emptyCheckBox}
        </div>
      );
      break;
    case 2:
      chain = (
        <div>
          {checkBox}
          {checkBox}
          {emptyCheckBox}
        </div>
      );
      break;
    case 3:
      chain = (
        <div>
          {checkBox}
          {checkBox}
          {checkBox}
        </div>
      );
      break;

    default:
      break;
  }

  return (
    <Box mt={2}>
      <Paper elevation={3} className="game-field">
        <Box mt={1} mb={1}>
          <div className="game-status">
            {levelBonus === 4 ? checkBox : chain}
            <Typography variant="body2" component="p">{`+ ${
              2 ** (levelBonus - 1) * 10
            } очков за слово`}</Typography>
          </div>
          <Typography variant="h5" component="p">
            {level}
          </Typography>
        </Box>
        <Divider />
        <Box mt={1} mb={1}>
          <Typography variant="h5" component="p">
            {currentWord}
          </Typography>
          <Typography variant="h5" component="p">
            {currentTranslate}
          </Typography>
        </Box>
        <Divider />
        <Box className="game-button-container">
          <Button
            size="large"
            variant="contained"
            color="secondary"
            onClick={(e) => onAnswerClick(e)}
          >
            Неверно
          </Button>
          <Button
            size="large"
            variant="contained"
            color="primary"
            onClick={(e) => onAnswerClick(e)}
          >
            Верно
          </Button>
        </Box>
      </Paper>
    </Box>
  );
};

export default GameField;
