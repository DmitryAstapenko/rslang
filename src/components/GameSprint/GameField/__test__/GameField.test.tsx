import React from "react";
import GameField from "../GameField";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

interface IGameFieldProps {
  currentWord: string;
  levelBonus: number;
  currentTranslate: string;
  onAnswerClick: (e: React.MouseEvent<HTMLButtonElement>) => void;
  currentChain: number;
}


const setUp = (props:IGameFieldProps) => shallow(<GameField {...props} />);

describe("should render Score component", () => {
  let component:any;
  beforeEach(() => {
    const currentWord: string = "main";
    const levelBonus: number = 0;
    const currentTranslate: string = "главный";
    const currentChain: number = 0;
    const onAnswerClick = (e : React.MouseEvent<HTMLButtonElement>) => {
      console.log(e);
    };
    component = setUp({levelBonus, currentWord, currentChain,currentTranslate,onAnswerClick});
  });
  it("should contain game-status block", () => {
    const wrapper = component.find(".game-status");
    expect(wrapper.length).toBe(1);
  });
  it("should render Score component", () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
