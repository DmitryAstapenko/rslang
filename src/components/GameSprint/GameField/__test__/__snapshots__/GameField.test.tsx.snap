// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Score component should render Score component 1`] = `
<WithStyles(ForwardRef(Card))
  className="game-field"
>
  <WithStyles(ForwardRef(CardContent))>
    <div
      className="game-status"
    >
      <div>
        <span>
          <Memo(ForwardRef(CheckBoxOutlineBlankIcon)) />
        </span>
        <span>
          <Memo(ForwardRef(CheckBoxOutlineBlankIcon)) />
        </span>
        <span>
          <Memo(ForwardRef(CheckBoxOutlineBlankIcon)) />
        </span>
      </div>
      <p>
        + 5 очков за слово
      </p>
    </div>
    <WithStyles(ForwardRef(Typography))
      component="h2"
      gutterBottom={true}
      variant="h5"
    >
      <Memo(ForwardRef(LooksOneIcon)) />
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      component="h2"
      gutterBottom={true}
      variant="h5"
    >
      main
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      component="h2"
      gutterBottom={true}
      variant="h5"
    >
      главный
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(CardContent))>
  <WithStyles(ForwardRef(CardActions))
    className="game-button-container"
  >
    <Styled(MuiBox)>
      <WithStyles(ForwardRef(Paper))
        elevation={2}
      >
        <Memo(ForwardRef(ArrowBackIcon)) />
      </WithStyles(ForwardRef(Paper))>
    </Styled(MuiBox)>
    <WithStyles(ForwardRef(Button))
      color="secondary"
      onClick={[Function]}
      variant="contained"
    >
      Неверно
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button))
      color="primary"
      onClick={[Function]}
      variant="contained"
    >
      Верно
    </WithStyles(ForwardRef(Button))>
    <Styled(MuiBox)
      ml={1}
    >
      <WithStyles(ForwardRef(Paper))
        elevation={2}
      >
        <Memo(ForwardRef(ArrowForwardIcon)) />
      </WithStyles(ForwardRef(Paper))>
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(CardActions))>
</WithStyles(ForwardRef(Card))>
`;
